siliconcompiler/core.py

```diff
@@ -2456,6 +2457,77 @@ def calc_dpw(self):
 
         return int(dies)
 
+
+    ###########################################################################
+    def check_logfile(self, step=None, jobname=None, index=None,
+                      logfile=None, display=True):
+        '''
+        Checks logfile for patterns found in the 'regex' parameter.
+
+        Reads the content of the step's log file and compares the
+        content found in step 'regex' parameter. The matches are
+        stored in the file 'reports/<design>.<suffix>' in the run directory.
+        The matches are printed to STDOUT if display is set to True.
+
+        Args:
+            step (str): Task step name ('syn', 'place', etc)
+            jobname (str): Jobid directory name
+            index (str): Task index
+            display (bool): If True, printes matches to STDOUT.
+
+        Examples:
+            >>> chip.check_logfile('place')
+            Searches for regex matches in the place logfile.
+        '''
+
+        # Using manifest to get defults
+        if step is None:
+            step = self.get('arg', 'step')
+        if index is None:
+            index = self.getkeys('flowgraph', step)[0]
+        if jobname is None:
+            jobname = self.get('jobname')
+        if logfile is None:
+            logfile = f"{step}.log"
+
+        tool = self.get('flowgraph', step, index, 'tool')
+        design = self.get('design')
+
+        # Creating local dictionary (for speed)
+        checks = {}
+
+        regex_list = []
+        if self.valid('eda', tool, 'regex', step, index, 'default'):
+            regex_list = self.getkeys('eda', tool, 'regex', step, index)
+
+        for suffix in regex_list:
+            checks[suffix] = {}
+            checks[suffix]['report'] = open(f"{step}.{suffix}", "w")
+            checks[suffix]['patterns'] = []
+            patterns = self.get('eda', tool, 'regex', step, index, suffix)
+            for item in patterns:
+                negate  = re.search(r'^-v (\w+)', item)
```

Comment:
Is there any chance that a negation regex could start with a non-word character such as `[]$%().`, etc? It looks like this will match a string like: `"-v error123"`, but not `"-v [error123]"`.

Reply:
> Is there any chance that a negation regex could start with a non-word character such as `[]$%().`, etc? It looks like this will match a string like: `"-v error123"`, but not `"-v [error123]"`.

Great catch!

Reply:
I went back and forth, I didn't like the hackiness of having a "not" operation, but clearly the -v will get us into tourble at some point, and is hacky! A less hacky compromise would be a tuple with (opt,regex), where opt  would be the switch given to grep (a subset). For example , -i, -x, -v, -w.,,,That feels right! Thoughts?

Reply:
That seems like a good compromise, especially if we plan to extend the functionality with more options. I'd be for that!